### ISSUE TITLE
Support NVDA table navigation in Microsoft Loop tables

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/nvda#16911]
### Is your feature request related to a problem? Please describe.
Microsoft Loop is an online service that allows the creation of components which can be included in Microsoft Office documents, or in a specific Loop workspace.
The Microsoft Loop canvas is implemented as a contenteditable div. 
Accessibility of Loop is generally okay, as NVDA already has fairly good support for reading and interacting within contenteditables, either in Chromium or Gecko.
One of the standard Loop components is a table, which contains column headers, and rows of data.
Generally, the table component reads okay, the user can certainly use the arrow keys to move between the cells and the content.  However, NVDA's table navigation commands (control + alt + arrows) are unavailable, and some users have an expectation that they should be in this scenario.

It is worth noting that NVDA does support its table navigation for tables in contenteditables generally, but not for these specific tables for the reason that these tables are specifically marked as non-contenteditable, even though they are within a contenteditable.
To be specific:
* The Loop canvas is contentedible.  
* The table, its rows and cells are non-contenteditable. 
* Each cell contains its own contenteditable div, which gets focus when the user is editing / navigating content inside the cell.

From NVDA's point of view, because the table is non-contenteditable, and because the inner contenteditable div in the cell specifically gets focus, NvDA treats this inner div as the edit area. And as this div does not itself contain any tables, then NvDA does not offer its table navigation commands.

### Describe the solution you'd like
When focused within a cell in a Loop table, NVDA's table navigation commands should allow the user to move focus to adjacent cells.

### Describe alternatives you've considered
None at this time.

### Additional context
NVDA so far has only supported its table navigation commands in two scenarios. These are within a text area with a caret, or a list control with multiple columns. As NvDA has never supported table navigation generically on tables with cells that receive focus, the addition of support for tables in Loop is a requested feature rather than a bug.

#### Steps to reproduce

Minimal testcase: 
[test_noncontenteditable_table.html.txt](https://github.com/user-attachments/files/16371618/test_noncontenteditable_table.html.txt)

Or in Loop:
* Visit https://loop.microsoft.com/
* Create a new workspace
* When focused in the canvas, press `/` and choose Table from the list to insert a table.

To reproduce, in either the minimal testcase or Loop:
* Ensure NVDA is in focus mode
* Arrow down into one of the table cells
* Try pressing an NVDA table navigation command E.g. `control+alt+rightArrow`. 

#### Expected behaviour:
Focus should move into the adjacent cell. NVDA should announce the focus.

#### Actual results:
NVDA reports that it is not in a table.

#### Possible implementation
`source/documentBase.py`, `DocumentWithTableNavigation._tableFindNewCell`:
If `_getTableCellCoordsCached` raises LookupError, then 
Do the following:
```
			parent = self.parent
			while parent:
				try:
					outerSelection = parent.makeTextInfo(self)
				except LookupError:
					outerSelection = None
				if outerSelection:
					outerSelection.collapse()
					return parent._tableFindNewCell(movement, axis, outerSelection, raiseOnEdge)
				parent = parent.parent
			else:
				# Translators: The message reported when a user attempts to use a table movement command
				# when the cursor is not within a table.
				ui.message(_("Not in a table cell"))
				raise e
```

This will loop through the ancestor objects, locating the focus object where we can create a TextInfo positioned at the original edit area's NVDAObject. In other words, locate this cell in the ancestor table.
Then find the new cell on the ancestor table.

In the minimal testcase, this successfully finds the ajacent table cell and moves focus there. However, NVDA reporting is very verbose, announcing like it has entered a new table. In theory the speakTextInfo call outside of where`_tableFindNewCell` was called can be suppressed in this scenario, as NVDA's normal focus reporting is enough.
  
However, When testing with Loop tables in Firefox at least, this implementation seems to cause focus to then jump to the document, which is not good. this may be a bug in Loop and or Firefox.
Further investigation is needed.

**Comment by msftedad** [comment]
Hi Team, Any update on this issue?

**Comment by seanbudd** [comment]
No updates

**Comment by Adriani90** [comment]
@michaelDCurran, @SaschaCowley is it possible to revisit this one?
1. Loops are getting very popular while e.g. using MS Teams in worksing environment, remote working in in such cases quite inefficient. At least in my case, people at my job use frequently tables in loop componnents
2. Table cells are very verbose in contentEditable, everytime NVDA reports "content editable" when landing on a cell. Is it possible to add a checkbox to document formatting settings or so to turn on and off announcement of "content editable"?
